app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -225,13 +225,23 @@ public TrackedEntitySearchCollectionRepository getFilteredRepository(SearchParam
                 TrackedEntityAttribute attribute = d2.trackedEntityModule().trackedEntityAttributes().uid(dataId).blockingGet();
                 boolean isUnique = attribute.unique();
                 boolean isOptionSet = (attribute.optionSet() != null);
-                if (isUnique || isOptionSet) {
-                    trackedEntityInstanceQuery = trackedEntityInstanceQuery.byFilter(dataId).eq(dataValue);
-                } else if (dataValue.contains("_os_")) {
-                    dataValue = dataValue.split("_os_")[1];
-                    trackedEntityInstanceQuery = trackedEntityInstanceQuery.byFilter(dataId).eq(dataValue);
-                } else
-                    trackedEntityInstanceQuery = trackedEntityInstanceQuery.byFilter(dataId).like(dataValue);
+                if(dataValues != null && dataValues.size()>1) {
+                    trackedEntityInstanceQuery = trackedEntityInstanceQuery.byFilter(dataId).in(dataValues);
+
+                } else {
+                    assert dataValues != null;
```

Comment:
you are asserting twice that `dataVaues != null`, here and in [line 228](https://github.com/dhis2/dhis2-android-capture-app/pull/4318/files#diff-ea895a67063b379e5a1b677889df995f59fd5333c349527593ee3a5a5007b03dR228) , maybe we can simplify this block.